submissions/AnnaStoyano/oop-exercise/index.js

```diff
@@ -0,0 +1,101 @@
+   Web app: _put project's github pages URL here_
+   */
+
+// ======== OBJECTS DEFINITIONS ========
+// Define your objects here
+
+class Mammal {
+   constructor(species, name, gender, legs, friends, saying) {
+      this.species = species;
+      this.name = name;
+      this.gender = gender;
+      this.legs = legs;
+      this.friends = friends;
+      this.saying = saying;
+   }
+
+   tellAbout() {
+      const properties = ['species', 'name', 'gender']; //all mammals have this properties
+      for (let property in this) {
+         if (!properties.includes(property))
+            properties.push(property);
+      }
+      return properties.map(property => `<b>${property}:</b> ${this[property]}`).join('\t');
+   }
+};
+
+class Dog extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('dog', name, gender, 4, friends, saying || 'guf');
+   }
+};
+
+class Cat extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('cat', name, gender, 4, friends, saying || 'meow');
+   }
+
+   getSaying() {
+      return this.saying || 'meow';
+   }
+}
+
+class Human extends Mammal {
+   constructor(name, gender, friends, saying) {
+      super('human', name, gender, 2, friends, saying);
+      this.hands = 2;
+      this.saying = saying || `Hello! My name is ${this.name}`;
+   }
+}
+
+class Woman extends Human {
+   constructor(name, friends, saying) {
+      super(name, 'female', friends, saying);
+   }
+}
+
+class Man extends Human {
+   constructor(name, friends, saying) {
+      super(name, 'male', friends, saying);
+   }
+}
+
+class CatWoman extends Woman {
+   constructor(name, friends) {
+      super(name, friends, Cat.prototype.getSaying());
+   }
+};
+
+const cat = new Cat('Murzic', 'male', ['Elza', 'Nicky']);
+const dog = new Dog('Lucy', 'female', ['Alla', 'Dima'], 'guffyy');
+const man = new Man('Vlad', ['Sergay', 'Anton'], 'Hello everyone!');
+const woman = new Woman('Anna', ['Sergay', 'Artem']);
+const catWoman = new CatWoman('Lily', ['Batman']);
+
+const inhabitants = [man, woman, cat, dog, catWoman];
+
+let output = inhabitants.map(inhabitant => inhabitant.tellAbout()).join('\n');
+
+print(output);
+// ======== OUTPUT ========
+/* Use print(message) for output.
+   Default tag for message is <pre>. Use print(message,'div') to change containing element tag.
+
+   Message can contain HTML markup. You may also tweak index.html and/or styles.css.
+   However, please, REFRAIN from improving visuals at least until your code is reviewed
+   so code reviewers might focus on a single file that is index.js.
+   */
+
+/* Print examples:
+   print('ABC');
+   print('<strong>ABC</strong>');
+   print('<strong>ABC</strong>', 'div');
+
+   print('human; John; male; 2; 2; Hello world!; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny');
+   print('human; <strong>John</strong>; male; 2; 2; <em>Hello world!</em>; Rex, Tom, Jenny', 'div');
+   */
```

Comment:
The mark below means that there is no newline character at the end of file.
![image](https://user-images.githubusercontent.com/8686631/99883109-cc706300-2c2d-11eb-94d6-e5d0994435dd.png)

You need to find a setting in your IDE that ensures every file ends with a newline character, or ensure having a newline  at every file manually.

[Reason why it is required and hints on how to fix it once forever](https://gist.github.com/OleksiyRudenko/d51388345ea55767b7672307fe35adf3)